packages/node_modules/@ciscospark/internal-plugin-metrics/src/metrics.js

```diff
@@ -99,6 +114,107 @@ const Metrics = SparkPlugin.extend({
       eventPayload: payload
     };
     return this.callDiagnosticEventsBatcher.request(event);
+  },
+
+  storeMetric(id, metricProps, metricType) {
+    const tuple = {metric: metricProps, type: metricType};
+    return this.metricMap.set(id, tuple);
+  },
+
+  getMetrics() {
+    return this.metricMap;
+  },
+
+  getMetric(id) {
+    return this.metricMap.get(id);
+  },
+
+  deleteMetric(id) {
+    return this.metricMap.delete(id);
+  },
+
+  registerMetric(eventName, id, props, type) {
+    if (!this.getMetric(id)) {
+      // hash name and props into table/object
+      const payload = this.createPayload(eventName, props);
+      return this.storeMetric(id, payload, type);
+    }
+    return false;
+  },
+
+  /**
+  * Registers a metric to be used to later update or call finish method
+  * @param {string} eventName name of the metric
+  * @param {string} id unique identifier
+  * @param {Object} props to populate the metric with
+  * @returns {bool} true if metric registered, false otherwise
+  */
+  clientRegisterMetric(eventName, id, props) {
+    return this.registerMetric(eventName, id, props, this.CLIENT_METRIC);
+  },
+
+  /**
+  * Registers a metric to be used to later update or call finish method
+  * @param {string} eventName name of the metric
+  * @param {string} id unique identifier
+  * @param {Object} props to populate the metric with
+  * @returns {bool} true if metric registered, false otherwise
+  */
+  sdkRegisterMetric(eventName, id, props) {
+    return this.registerMetric(eventName, id, props, this.SDK_METRIC);
+  },
+
+  /**
+   * Updates a registered metric, if it exists
+   * @param {string} id lookup key for metric
+   * @param {Object} metricProps to be used for updating metric
+   * @returns {bool} true if the metrics exists and was updated, false otherwise
+   */
+  updateRegisteredMetric(id, metricProps) {
+    const storedMetric = this.getMetric(id);
+    const props = storedMetric ? storedMetric.metric : null;
+    const type = storedMetric ? storedMetric.type : null;
+
+    if (!props || !type || !metricProps) {
+      return false;
+    }
+
+    // update metric
+    _.assign(props.fields, metricProps.fields);
```

Comment:
Use `Object.assign`. Generally, you want to avoid using lodash methods if there's a native method available.

Reply:
done